expr/func_invert_test.go

```diff
@@ -0,0 +1,140 @@
+package expr
+
+import (
+	"math"
+	"strconv"
+	"testing"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/schema"
+	"github.com/grafana/metrictank/test"
+)
+
+var datapoints = []schema.Point{
+	{Val: 0, Ts: 10},
+	{Val: math.NaN(), Ts: 20},
+	{Val: -10, Ts: 30},
+	{Val: 5.5, Ts: 40},
+	{Val: -math.MaxFloat64, Ts: 50},
+	{Val: -1234567890, Ts: 60},
+	{Val: math.MaxFloat64, Ts: 70},
+}
+
+var datapointsInvert = []schema.Point{
+	{Val: math.NaN(), Ts: 10},
+	{Val: math.NaN(), Ts: 20},
+	{Val: math.Pow(-10, -1), Ts: 30},
+	{Val: math.Pow(5.5, -1), Ts: 40},
+	{Val: math.Pow(-math.MaxFloat64, -1), Ts: 50},
+	{Val: math.Pow(-1234567890, -1), Ts: 60},
+	{Val: math.Pow(math.MaxFloat64, -1), Ts: 70},
+}
+
+func TestInvert(t *testing.T) {
+	in := []models.Series{
+		{
+			Interval:   10,
+			QueryPatt:  "queryPattHere",
+			Target:     "targetHere",
+			Datapoints: getCopy(datapoints),
+		},
+	}
+
+	// store copy of the original input
+	inCopy := []models.Series{
```

Comment:
Did you verify that this failed before the fix?

Reply:
Yes, all the fields were getting modified in-place.